openff/toolkit/_tests/test_topology.py

```diff
@@ -144,6 +144,38 @@ def test_empty(self):
         assert not topology.is_periodic
         assert len(topology.constrained_atom_pairs.items()) == 0
 
+    def test_from_molecule_bad_argument(self):
+        with pytest.raises(
+            ValueError,
+            match="Invalid type.*Topology",
+        ):
+
+            topology = Topology()
+
+            topology.add_molecule(create_water().to_topology())
+
+    @pytest.mark.timeout(8)
+    def test_from_molecules(self):
+        water = create_water()
+
+        topology = Topology()
+
+        indices = topology.add_molecules(10_000 * [water])
+
+        assert topology.n_molecules == 10_000
+
+        assert indices == [*range(1, 10_001)]
```

Comment:
I think this should be 
```suggestion
        assert indices == [*range(10_000)]
```
but the behavior (of both functions) is to return a 1-indexed index whereas I expected 0-indexed:  #1922

Reply:
(not blocking) Agree, please feel free to change this behavior for `add_molecule` and `add_molecules` in this PR (though you've already gone through one major change so we can also handle this in a future PR if you prefer).